src/framework/fixture.ts

```diff
@@ -58,33 +54,38 @@ export class Fixture {
     return ret;
   }
 
-  protected eventualAsyncExpectation<T>(fn: () => Promise<T>): Promise<T> {
-    const promise = fn();
+  protected eventualAsyncExpectation<T>(fn: (niceStack: Error) => Promise<T>): Promise<T> {
+    const promise = fn(new Error());
     this.eventualExpectations.push(promise);
     return promise;
   }
 
-  private expectErrorValue(expectedName: string, ex: unknown, m: string): void {
+  private expectErrorValue(expectedName: string, ex: unknown, niceStack: Error): void {
```

Comment:
Is `niceStack` because we can't/don't want to use `stack`?

Reply:
No, just because it's more descriptive. I.e. it's the "nice stack" to show alongside this message instead of the "real" stack (because it's captured before any awaits).

Reply:
(I'll revise after landing if you think another name would be better.)